src/app/applications/multicast/multicast-table/multicast-table.component.ts

```diff
@@ -0,0 +1,134 @@
+  @Input() pageLimit: number;
+  public pageOffset = 0;
+  public pageTotal: number;
+  public applicationId: number;
+
+  private multicastSubscription: Subscription;
+  private deleteDialogSubscription: Subscription;
+
+  constructor(
+    private route: ActivatedRoute,
+    private deleteDialogService: DeleteDialogService,
+    private multicastService: MulticastService,
+    private meService: MeService,
+    public translate: TranslateService,
+    public snackService: SnackService
+  ) {
+    translate.use('da');
+  }
+
+  ngOnInit(): void {
+    this.applicationId = +Number(this.route.parent.snapshot.paramMap.get('id'));
+    this.getMulticasts(); // loads the multicasts
+    this.canEdit = this.meService.canWriteInTargetOrganization();
+  }
+
+  ngAfterViewInit() {
+    this.dataSource.paginator = this.paginator;
+    this.dataSource.sort = this.sort;
+  }
+
+  getMulticasts(): void {
+    if (this.applicationId) {
+      this.multicastSubscription = this.multicastService
+        .getMulticastsByApplicationId(
+          // gets multicasts from db by applicationId
+          this.pageLimit,
+          this.pageOffset * this.pageLimit,
+          this.applicationId
+        )
+        .subscribe((multicasts: MulticastData) => {
+          this.multicasts = multicasts.data;
+          this.dataSource = new MatTableDataSource<Multicast>(this.multicasts); // these lines of code is inspired/taken from datatarget.
+          this.dataSource.paginator = this.paginator;
+          this.dataSource.sort = this.sort;
+          this.dataSource.sortingDataAccessor = tableSorter;
+          this.isLoadingResults = false;
+          if (this.pageLimit) {
+            this.pageTotal = Math.ceil(multicasts.count / this.pageLimit);
+          }
+          if (multicasts.ok === false) {
+            this.showLoadFailSnack(); // if not possible to load the multicasts, show error.
+          }
+        });
+    }
+  }
+
+  deleteMulticast(multicast: Multicast) {
+    this.deleteDialogSubscription = this.deleteDialogService
+      .showSimpleDialog()
+      .subscribe((response) => {
+        if (response) { // if user presses "yes, delete", then delete the multicast.
+          this.multicastService
+            .delete(multicast.id)
+            .subscribe((response) => {
+              if (response.ok && response.body.affected > 0) { // if deleted succesfully, get the new array of multicasts and show a succesful snack.
+                this.getMulticasts();
+                this.showDeletedSnack();
+              } else {
+                this.showFailSnack();
+              }
+            });
+        } else {
+          console.log(response);
+        }
+      });
+  }
+
+  showLoadFailSnack() {
+    this.snackService.showLoadFailSnack();
+  }
+  showFailSnack() {
+    this.snackService.showFailSnack();
+  }
+  showDeletedSnack() {
+    this.snackService.showDeletedSnack();
+  }
+  ngOnDestroy() { // inspired by datatarget
```

Comment:
No need for this type of comment unless the code is heavily inspired by an article or forum post.